language/solana/move-to-solana/src/stackless/dwarf.rs

```diff
@@ -771,28 +1004,132 @@ impl<'up> DIBuilder<'up> {
                 )
             };
 
-            let llvm_builder = &mod_cx.llvm_builder;
+            let module_context = unsafe { LLVMGetModuleContext(module.0) };
+            let debug_location = unsafe {
+                LLVMDIBuilderCreateDebugLocation(
+                    module_context,
+                    lineno,
+                    column,
+                    /* Scope */ lexical_block,
+                    std::ptr::null_mut(), // Inlined at
+                )
+            };
 
-            let entry_bb = llvm_builder.get_entry_basic_block(ll_fn);
-            dbg!(entry_bb);
+            unsafe {
+                LLVMSetSubprogram(ll_fn.0, function);
+            };
 
-            let module_di = &self.module_di().unwrap();
-            let module_ctx = unsafe { LLVMGetModuleContext(*module_di) };
+            let llvm_builder = &module_cx.llvm_builder;
+
+            let _entry_bb = llvm_builder.get_entry_basic_block(ll_fn);
+
+            // 0. No need to set compiled_unit, it was set in new()
```

Comment:
I spent significant effort selecting the right params on each step, and while I was playing the experiments the numbers were very convenient for referencing. The order and selected parameters are correct but I'm not sure they are the best, and while I plan to keep finding the best params I want to have the system of enumeration in place. Once it is over it can be removed.
I added a long comment explaining this.